### DISCUSSION TITLE
initscripts directory is empty

### DISCUSSION THREAD

**Discussion by RahulARanger** [marcelotduarte/cx_Freeze#1037] [Q&A] [answered]
When I tried to install the cx_Freeze through PIP it gave me an Empty init scripts directory due to which I get an exception 
`No initscript named Console`.

I found a similar issue in [SO](https://stackoverflow.com/questions/15862761/no-initscrip-name-console-cx-freeze), but that was unsolved, I didn't raise this issue since this seems to be something related to the PIP, is there any way to resolve this issue?

**Comment by marcelotduarte** [comment, marked as answer]
Check if your `pip --version` return correct value for your python
What OS is your? Ubuntu, Windows, macOS, etc
Try to downgrade pip: python -m pip install "pip<21.1"
Then reinstall cx-freeze: pip install --force --no-cache cx-freeze

**Reply by RahulARanger** [reply]
I am using windows, and my pip version is 21.0.1.

I was trying to freeze an app along with pip hoping that one could produce lightweight exe with cx_Freeze.
I tried to install PySide and it succeeded, but when I tried cx_Freeze it did install but the initscript directory was empty

**Reply by marcelotduarte** [reply]
What python version? python -VV

**Reply by RahulARanger** [reply]
`python 3.8.8`

**Reply by marcelotduarte** [reply]
result of python -VV?
Have you tried the pip force command?
pip install --force --no-cache cx-freeze

**Reply by RahulARanger** [reply]
sorry for late response,
after trying `python -VV` it gave me `Python 3.8.8 (tags/v3.8.8:024d805, Feb 19 2021, 13:18:16) [MSC v.1928 64 bit (AMD64)]`

`pip install --force --no-cache cx-freeze` has made the app download cx_Freeze when ever I open the exe but it still misses the initscripts folder

**Reply by RahulARanger** [reply]
I was trying to make the cx_Freezed app dynamically download the packages using pip

so this was one of my samples

setup.py

```python
from cx_Freeze import setup, Executable
import pip, distutils
import pathlib
import sys

sys.argv.append("build")

options = {
    "build_exe": {
        "excludes": ["pip", "distutils"],
        "include_files": [
            (pathlib.Path(pip.__file__).parent, "pip"),
            (pathlib.Path(distutils.__file__).parent, "distutils")
        ],
    },

}
arguments = {
    "name": "test",
    "options": options,
    "executables": [Executable(target_name="test_pip", base="Console", script="main.py")],
    "version": "0.0.1"

}

setup(
    **arguments
)
```

main.py

```python
import os

while True:
    check = input()
    os.system(check)
    if not check:
        break

```

This can run some python commands but while doing pip commands I use a target to point towards the lib directory

But when I try to download `cx_Freeze` it did download successfully.
 but I think, cx_Freeze could not be able to convert the files inside its [initscripts](https://github.com/marcelotduarte/cx_Freeze/tree/main/cx_Freeze/initscripts) into `.pyc` files.
so the app couldn't able to use cx_Freeze.

**Reply by marcelotduarte** [reply]
See my comment at #992

**Reply by RahulARanger** [reply]
Thank you, Now I understand.
